python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -316,13 +246,145 @@ def _computePsfStats(self, sources):
+                location=loc,
+            )
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore')
+                altaz = coord.transform_to(AltAz)
+
+            summary.zenithDistance = float(90.0 - altaz.alt.degree)
+
+    def update_photo_calib_stats(self, summary, photo_calib):
+        """Compute all summary-statistic fields that depend on the photometric
+        calibration model.
+
+        Parameters
+        ----------
+        summary : `ExposureSummaryStatistics`
+            Summary object to update in-place.
+        photo_calib : `lsst.afw.image.PhotoCalib` or `None`
+            Photometric calibration model.  If `None`, all fields that depend
+            on the photometric calibration will be reset (generally to NaN).
+        """
+        if photo_calib is not None:
+            summary.zeroPoint = float(2.5*np.log10(photo_calib.getInstFluxAtZeroMagnitude()))
+        else:
+            summary.zeroPoint = float("nan")
+
+    def update_background_stats(self, summary, background):
+        """Compute summary-statistic fields that depend on only on the
+        background model.
+
+        Parameters
+        ----------
+        summary : `ExposureSummaryStatistics`
+            Summary object to update in-place.
+        background : `lsst.afw.math.BackgroundList` or `None`
+            Background model.  If `None`, all fields that depend
+            on the background will be reset (generally to NaN).
+
+        Notes
+        -----
+        This does not include fields that depend on the background-subtracted
+        masked image; when the background changes, it should generally be
+        applied to the image and `update_masked_image_stats` should be called
+        as well.
+        """
+        if background is not None:
+            bgStats = (bg[0].getStatsImage().getImage().array
+                       for bg in background)
+            summary.skyBg = float(sum(np.median(bg[np.isfinite(bg)]) for bg in bgStats))
+        else:
+            summary.skyBg = float("nan")
+
+    def update_masked_image_stats(self, summary, masked_image):
+        """Compute summary-statistic fields that depend on the masked image
+        itself.
+
+        Parameters
+        ----------
+        summary : `ExposureSummaryStatistics`
+            Summary object to update in-place.
+        masked_image : `lsst.afw.image.MaskedImage` or `None`
+            Masked image.  If `None`, all fields that depend
+            on the masked image will be reset (generally to NaN).
+        """
+        nan = float("nan")
+        if masked_image is None:
+            summary.skyNoise = nan
+            summary.meanVar = nan
+            return
+        statsCtrl = afwMath.StatisticsControl()
+        statsCtrl.setNumSigmaClip(self.config.sigmaClip)
+        statsCtrl.setNumIter(self.config.clipIter)
+        statsCtrl.setAndMask(afwImage.Mask.getPlaneBitMask(self.config.badMaskPlanes))
```

Comment:
Did you also want to update the entries in `self.config.badMaskPlanes` here, or is that another ticket?

Reply:
I think it's another ticket.  I don't this one to rock any unnecessary boats.